h/static/scripts/directive/test/annotation-test.js

```diff
@@ -0,0 +1,1377 @@
+    $provide.value('momentFilter', fakeMomentFilter);
+    $provide.value('permissions', fakePermissions);
+    $provide.value('personaFilter', fakePersonaFilter);
+    $provide.value('documentTitleFilter', fakeDocumentTitleFilter);
+    $provide.value('documentDomainFilter', fakeDocumentDomainFilter);
+    $provide.value('session', fakeSession);
+    $provide.value('store', fakeStore);
+    $provide.value('tags', fakeTags);
+    $provide.value('time', fakeTime);
+    $provide.value('urlencodeFilter', fakeUrlEncodeFilter);
+    $provide.value('groups', fakeGroups);
+  }));
+
+  beforeEach(
+    inject(
+      function(_$compile_, _$document_, _$q_, _$rootScope_, _$timeout_,
+               _$window_) {
+        $compile = _$compile_;
+        $document = _$document_;
+        $window = _$window_;
+        $q = _$q_;
+        $timeout = _$timeout_;
+        $rootScope = _$rootScope_;
+        $scope = $rootScope.$new();
+        $scope.annotation = annotation = {
+          id: 'deadbeef',
+          document: {
+            title: 'A special document'
+          },
+          target: [{}],
+          uri: 'http://example.com',
+          user: 'acct:bill@localhost'
+        };
+      }
+    )
+  );
+
+  afterEach(function() {
+    sandbox.restore();
+  });
+
+  describe('when the annotation is a highlight', function() {
+    beforeEach(function() {
+      annotation.$highlight = true;
+      annotation.$create = sinon.stub().returns({
+        then: angular.noop,
+        'catch': angular.noop,
+        'finally': angular.noop
+      });
+    });
+
+    it('persists upon login', function() {
+      delete annotation.id;
+      delete annotation.user;
+      fakeSession.state.userid = null;
+      createDirective();
+      $scope.$digest();
+      assert.notCalled(annotation.$create);
+      fakeSession.state.userid = 'acct:ted@wyldstallyns.com';
+      $scope.$broadcast(events.USER_CHANGED, {});
+      $scope.$digest();
+      assert.calledOnce(annotation.$create);
+    });
+
+    it('is private', function() {
+      delete annotation.id;
+      createDirective();
+      $scope.$digest();
+      assert.deepEqual(annotation.permissions, {
+        read: ['justme']
+      });
+    });
+  });
+
+  describe('#reply', function() {
+    var container;
+
+    beforeEach(function() {
+      createDirective();
+      annotation.permissions = {
+        read: ['acct:joe@localhost'],
+        update: ['acct:joe@localhost'],
+        destroy: ['acct:joe@localhost'],
+        admin: ['acct:joe@localhost']
+      };
+    });
+
+    it('creates a new reply with the proper uri and references', function() {
+      var match;
+      controller.reply();
+      match = sinon.match({
+        references: [annotation.id],
+        uri: annotation.uri
+      });
+      assert.calledWith(fakeAnnotationMapper.createAnnotation, match);
+    });
+
+    it('makes the annotation shared if the parent is shared', function() {
+      var reply;
```

Comment:
Again, I'm guessing this comes from the CoffeeScript compiler but I'd suggest going through and merging definitions and declarations where this is easy to do.